kayenta-canary-analysis/src/main/java/com/netflix/kayenta/canaryanalysis/domain/CanaryAnalysisExecutionRequestScope.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018 Nike, inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.canaryanalysis.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+@Data
+@Builder
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+@ApiModel(description="Defines how to find the metrics for the experiment and control in the metrics source")
+public class CanaryAnalysisExecutionRequestScope {
+
+  @Builder.Default
+  @ApiModelProperty(
+      value = "TODO: someone who uses this, should update this",
+      example = "default")
+  String scopeName = "default";
+
+  @ApiModelProperty(
+      value = "Value used in the metric source metrics service to fetch the data for the control. " +
+          "The ASG name is the default value supplied via the Spinnaker UX",
+      example = "examplemicroservice--control-v001")
+  String controlScope;
+
+  @ApiModelProperty(
+      value = "The location of the control, used by some metrics sources to further differentiate metrics",
+      example = "us-west-2")
+  String controlLocation;
+
+  @ApiModelProperty(
+      value = "Value used in the metric source metrics service to fetch the data for the experiment. " +
+          "The ASG name is the default value supplied via the Spinnaker UX",
+      example = "examplemicroservice--experiment-v001")
+  String experimentScope;
+
+  @ApiModelProperty(
+      value = "The location of the experiment, used by some metrics sources to further differentiate metrics.",
+      example = "us-west-2")
+  String experimentLocation;
+
+  @ApiModelProperty(
+      value = "If supplied this value will be used instead of the current time at execution for looking up metrics.",
+      example = "2018-12-17T20:56:39.689Z"
+  )
+  String startTimeIso;
```

Comment:
@duftler is there an action item for this, or is this just to educate me on the reason for its original existence.

Reply:
Sorry, should have provided more details. I mentioned this here because your interface looked to me like it wouldn't support that use case (since there is only one set of time boundaries exposed). This is from memory; maybe I missed something in the interface?

Reply:
- [ ] investigate and address

Reply:
@duftler ok, my memory is refreshed, this is a thing that I ported from the [orca integration](https://github.com/spinnaker/orca/blob/master/orca-kayenta/src/main/kotlin/com/netflix/spinnaker/orca/kayenta/model/KayentaCanaryContext.kt#L66) without much thought on optimizing or refactoring.

Thinking about it know, I can update the API / models to follow what the Canary controller does and take a map of scope names to CanaryScopePairs rather than having the flat scope object that lacks the functionality that you are referencing in your comments.

How does that sound?

Reply:
Following up on our offline discussion, I think it's ok to leave it as-is for now. If there is a lot of interest the described use case, we can always add support.

Reply:
- [X] done